scene/gui/menu_button.cpp

```diff
@@ -49,199 +109,80 @@ void MenuButton::shortcut_input(const Ref<InputEvent> &p_event) {
-				popup->hide();
-			}
-		} break;
-
-		case NOTIFICATION_INTERNAL_PROCESS: {
-			Vector2i mouse_pos = DisplayServer::get_singleton()->mouse_get_position() - mouse_pos_adjusted;
-			MenuButton *menu_btn_other = Object::cast_to<MenuButton>(get_viewport()->gui_find_control(mouse_pos));
-
-			if (menu_btn_other && menu_btn_other != this && menu_btn_other->is_switch_on_hover() && !menu_btn_other->is_disabled() &&
-					(get_parent()->is_ancestor_of(menu_btn_other) || menu_btn_other->get_parent()->is_ancestor_of(popup))) {
-				popup->hide();
-				menu_btn_other->pressed();
-			}
-		} break;
+	PopupMenu *pm = get_popup();
+	if (!pm) {
+		return;
 	}
-}
 
-bool MenuButton::_set(const StringName &p_name, const Variant &p_value) {
-	Vector<String> components = String(p_name).split("/", true, 2);
-	if (components.size() >= 2 && components[0] == "popup") {
-		bool valid;
-		popup->set(String(p_name).trim_prefix("popup/"), p_value, &valid);
-		return valid;
-	}
-	return false;
+	pm->set_item_count(p_count);
+	notify_property_list_changed();
 }
 
-bool MenuButton::_get(const StringName &p_name, Variant &r_ret) const {
-	Vector<String> components = String(p_name).split("/", true, 2);
-	if (components.size() >= 2 && components[0] == "popup") {
-		bool valid;
-		r_ret = popup->get(String(p_name).trim_prefix("popup/"), &valid);
-		return valid;
+int MenuButton::get_item_count() const {
+	PopupMenu *pm = get_popup();
+	if (!pm) {
+		return 0;
 	}
-	return false;
-}
-
-void MenuButton::_get_property_list(List<PropertyInfo> *p_list) const {
-	for (int i = 0; i < popup->get_item_count(); i++) {
-		p_list->push_back(PropertyInfo(Variant::STRING, vformat("popup/item_%d/text", i)));
-
-		PropertyInfo pi = PropertyInfo(Variant::OBJECT, vformat("popup/item_%d/icon", i), PROPERTY_HINT_RESOURCE_TYPE, "Texture2D");
-		pi.usage &= ~(popup->get_item_icon(i).is_null() ? PROPERTY_USAGE_STORAGE : 0);
-		p_list->push_back(pi);
-
-		pi = PropertyInfo(Variant::INT, vformat("popup/item_%d/checkable", i), PROPERTY_HINT_ENUM, "No,As Checkbox,As Radio Button");
-		pi.usage &= ~(!popup->is_item_checkable(i) ? PROPERTY_USAGE_STORAGE : 0);
-		p_list->push_back(pi);
-
-		pi = PropertyInfo(Variant::BOOL, vformat("popup/item_%d/checked", i));
-		pi.usage &= ~(!popup->is_item_checked(i) ? PROPERTY_USAGE_STORAGE : 0);
-		p_list->push_back(pi);
-
-		pi = PropertyInfo(Variant::INT, vformat("popup/item_%d/id", i), PROPERTY_HINT_RANGE, "0,10,1,or_greater");
-		p_list->push_back(pi);
 
-		pi = PropertyInfo(Variant::BOOL, vformat("popup/item_%d/disabled", i));
-		pi.usage &= ~(!popup->is_item_disabled(i) ? PROPERTY_USAGE_STORAGE : 0);
-		p_list->push_back(pi);
-
-		pi = PropertyInfo(Variant::BOOL, vformat("popup/item_%d/separator", i));
-		pi.usage &= ~(!popup->is_item_separator(i) ? PROPERTY_USAGE_STORAGE : 0);
-		p_list->push_back(pi);
-	}
+	return pm->get_item_count();
 }
 
 void MenuButton::_bind_methods() {
-	ClassDB::bind_method(D_METHOD("get_popup"), &MenuButton::get_popup);
```

Comment:
I'm guessing we can't expose this method with a more specific return type in the derived class?

Reply:
I don't think so.